kentik_api_library/tests/e2e/cloudexport/test_aws.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+import pytest
+from requests import patch
+
+from kentik_api.cloudexport.cloud_export import AwsProperties, CloudExport, CloudExportType
+from kentik_api.public.types import ID
+from tests.e2e.cloudexport.utils import client, credentials_missing_str, credentials_present
+
+
+@pytest.mark.skipif(not credentials_present, reason=credentials_missing_str)
+def test_aws_crud() -> None:
+    PLAN_ID = ID(os.environ["KTAPI_PLAN_ID"])
+    PROPERTIES1 = AwsProperties(
+        bucket="cloudexport-src-s3-bucket",
+        iam_role_arn="arn:aws:iam::177634097045:role/test-cloudexport-IngestRole",
+        region="eu-central-1",
+        delete_after_read=True,
+        multiple_buckets=True,
+    )
+    PROPERTIES2 = AwsProperties(
+        bucket="cloudexport-src-s3-bucket-updated",
+        iam_role_arn="arn:aws:iam::177634097045:role/test-cloudexport-IngestRole-updated",
+        region="us-east-1",
+        delete_after_read=False,
+        multiple_buckets=False,
+    )
+    AWS = CloudExport(
+        type=CloudExportType.CUSTOMER_MANAGED,
+        enabled=True,
+        name="e2e-aws-cloudexport",
+        description="E2E test AWS CloudExport description",
+        api_root="https://api.kentik.com",
+        flow_dest="https://flow.kentik.com",
+        plan_id=PLAN_ID,
+        cloud_provider="aws",
+        aws=PROPERTIES1,
+    )
+
+    try:
+        # create
+        created = client().cloud_export.create(AWS)
+        assert isinstance(created, CloudExport)
+        assert created.id != ID()
+        assert created.type == CloudExportType.CUSTOMER_MANAGED
```

Comment:
Test the expected values against the input attributes for the create call where applicable. So, for example here:
```python
assert created.type == AWS.type
assert created.enabled == AWS.enabled
assert created.name == AWS.name
...
```
This will make the tests much more reusable and less error prone in case of future modifications.

Reply:
Updating